### PR TITLE
Update teams_discussions with ByID and BySlug endpoints

### LINKED ISSUE #1424: Update teams_discussions.go
Relates to: #1422 and #1387.

It seems that the [original blog post](https://developer.github.com/changes/2020-01-21-moving-the-team-api-endpoints/) that we were basing #1387 on didn't mention these changes.

This issue will be for updating `teams_discussions.go`.

The work involved here is to update every endpoint listed here:
https://developer.github.com/v3/teams/discussions/
Note that #1400 can be used as a model because most endpoints have two ways to call them: `*ByID` and `*BySlug`. The same pattern should be followed for these updates.

This would be a great PR for any new contributor to this repo or a new Go developer.
All contributions are greatly appreciated!

Feel free to volunteer for any issue and the issue can be assigned to you so that others don't attempt to duplicate the work.

Please check out our [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) guide to get started.

Thank you!

### PR DESCRIPTION
**Pull request by kadern0**
Fixes #1424

### CODE REVIEW

File: github/teams_discussions.go [hunk 1]
```diff
@@ -44,12 +44,12 @@ type DiscussionListOptions struct {
...
 }
 
-// ListDiscussions lists all discussions on team's page.
+// ListDiscussionsByID lists all discussions on team's page given Organization and Team ID.
 // Authenticated user must grant read:discussion scope.
 //
 // GitHub API docs: https://developer.github.com/v3/teams/discussions/#list-discussions
-func (s *TeamsService) ListDiscussions(ctx context.Context, teamID int64, options *DiscussionListOptions) ([]*TeamDiscussion, *Response, error) {
-	u := fmt.Sprintf("teams/%v/discussions", teamID)
+func (s *TeamsService) ListDiscussionsByID(ctx context.Context, orgID, teamID int64, options *DiscussionListOptions) ([]*TeamDiscussion, *Response, error) {
```
Comment (gmlewis) [inline github/teams_discussions.go]: Let's please change `options` to `opts` to match what was recently done in #1417.
Here and throughout this PR.

File: github/teams_discussions.go [hunk 2]
```diff
@@ -44,12 +44,12 @@ type DiscussionListOptions struct {
 	Direction string `url:"direction,omitempty"`
```
Comment (gmlewis) [inline github/teams_discussions.go]: From the documentation, it looks like `DiscussionListOptions` also needs to embed a `ListOptions` struct to support pagination. Please add this.

Comment (kadern0) [reply github/teams_discussions.go]: That's correct, I've updated it.

**Comment by gmlewis** [comment]
@wesleimp - this could also be reviewed if you have time.

**Comment by gmlewis** [comment]
Thank you, @wesleimp!
Merging.

**Pull request merged**